bundles/org.openhab.binding.coolmasternet/src/main/java/org/openhab/binding/coolmasternet/internal/ControllerHandler.java

```diff
@@ -44,116 +47,163 @@
+            if (h != null) {
+                h.refresh();
             }
-        } catch (CoolMasterClientError e) {
-            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, e.getMessage());
+        }
+        if (isConnected()) {
+            updateStatus(ThingStatus.ONLINE);
+        } else {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
         }
     }
 
-    /*
-     * Return true if the client socket is connected.
+    /**
+     * Passively determine if the client socket appears to be connected, but do
+     * modify the connection state.
      *
-     * Use checkConnection() to probe if the coolmasternet is responding correctly,
-     * and try to re-establish the connection if possible.
+     * <p>
+     * Use {@link #checkConnection()} if active verification (and potential
+     * reconnection) of the CoolNetMaster connection is required.
      */
     public boolean isConnected() {
         synchronized (socketLock) {
-            Socket localSocket = socket;
-            return localSocket != null && localSocket.isConnected() && !localSocket.isClosed();
+            final Socket socket = this.socket;
+            return socket != null && socket.isConnected() && !socket.isClosed();
         }
     }
 
-    /*
-     * Send a particular ASCII command to the CoolMasterNet, and return the successful response as a string.
+    /**
+     * Send a specific ASCII I/F command to CoolMasterNet and return its response.
+     *
+     * <p>
+     * This method automatically acquires a connection.
      *
-     * If the "OK" prompt is not received then a CoolMasterClientError is thrown that contains whatever
-     * error message was printed by the CoolMasterNet.
+     * @return the server response to the command (never empty)
+     * @throws {@link IOException} if communications failed with the server
      */
-    @SuppressWarnings("resource")
-    public @Nullable String sendCommand(String command) throws CoolMasterClientError {
+    public String sendCommand(final String command) throws IOException {
         synchronized (socketLock) {
             checkConnection();
 
-            StringBuilder response = new StringBuilder();
+            final StringBuilder response = new StringBuilder();
             try {
-                Socket localSocket = socket;
-                if (localSocket == null || !isConnected()) {
+                final Socket socket = this.socket;
+                if (socket == null || !isConnected()) {
                     throw new CoolMasterClientError(String.format("No connection for sending command %s", command));
                 }
 
                 logger.trace("Sending command '{}'", command);
-                OutputStream out = localSocket.getOutputStream();
-                out.write(command.getBytes(StandardCharsets.US_ASCII));
+                OutputStream out = socket.getOutputStream();
```

Comment:
If you are looking to make good general improvements to make, then you could wrap this output stream in a OutputStreamWriter so you can let the writer deal with Charset conversions for you. Should allow you to get rid of the `LF` constant you are using below.

Reply:
Do you mean `BufferedWriter(new OutputStreamWriter(socket.getOutputStream(), US_ASCII))` so you get `BufferedWriter.newLine()`?

Reply:
I don't think a BufferedWriter would be very useful in this case since the buffer is so short lived. Also it seems like '\n' was used pretty intentionally rather than a system newline so I think using `BufferedWriter.newLine()` might break things.
My suggestion is to just use `new OutputStreamWriter(socket.getOutputStream(), US_ASCII)`

Reply:
OK done.